src/OrchardCore/OrchardCore.DisplayManagement/Descriptors/DefaultShapeTableManager.cs

```diff
@@ -28,13 +29,13 @@ public class DefaultShapeTableManager : IShapeTableManager
         private static readonly object _syncLock = new();
 
         // Singleton cache to hold a tenant's theme ShapeTable 
-        private readonly Dictionary<string, ShapeTable> _shapeTableCache;
+        private readonly IDictionary<string, ShapeTable> _shapeTableCache;
 
         private readonly IServiceProvider _serviceProvider;
         private readonly ILogger _logger;
 
         public DefaultShapeTableManager(
-            [FromKeyedServices(nameof(DefaultShapeTableManager))] Dictionary<string, ShapeTable> shapeTableCache,
+            [FromKeyedServices(nameof(DefaultShapeTableManager))] IDictionary<string, ShapeTable> shapeTableCache,
```

Comment:
shouldn't this be `ConcurrentDictionary` instead?

Reply:
Declared as IDictionary, but registered as ConcurrentDictionary. Just in case we want to change the implementation without breaking the service contract. But risk is super low since this is the only usage. If it's too complex/ambiguous we can make it explicit.

Reply:
This is something is resolved on every request. It's possible that more than one thread tries to populate it at the same time.

Reply:
No, this is registered as a singleton, a single CD will be created.

Reply:
Understand. But 2 threads can request the same singleton instance and try to populate it at the same time.  Yes low risk, but the risk is there

Reply:
Not an issue at all. But right I could add a double lock for that case, good idea.